app/src/main/java/fr/tabbya/instantvisio/MainActivity.java

```diff
@@ -85,7 +84,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
         button = findViewById(R.id.button);
         button.setOnClickListener(v -> {
-            if (ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.SEND_SMS) != PackageManager.PERMISSION_GRANTED) {
+            if (simDevice&&ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.SEND_SMS) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
code style convention, spaces would give better readability here:

```java
simDevice && ContextCompat...
```

We should be able sharing the same formatting style, I'll research upon this